### PR TITLE
Type check first extension (amp-youtube)

### PR DESCRIPTION
**Pull request by cramforce**
- Compiles together with main binary. While this may overlook some problems, it avoid unknown type warnings.
- Adds a dedicated extern for `AMP.BaseElement`. This is needed to get the inheritance hierarchy correct and not get unknown type warnings for other compilation units.

### CODE REVIEW

File: extensions/amp-youtube/0.1/amp-youtube.js [hunk 1]
```diff
@@ -29,13 +29,34 @@ const YT_PLAYER_STATE_PLAYING = 1;
 
 class AmpYoutube extends AMP.BaseElement {
 
-  /** @override */
-  preconnectCallback(onLayout) {
-    this.preconnect.url('https://www.youtube.com', onLayout);
+  constructor() {
```
Comment (erwinmombay) [inline extensions/amp-youtube/0.1/amp-youtube.js]: do we need to start doing this for components?

Comment (erwinmombay) [inline extensions/amp-youtube/0.1/amp-youtube.js]: need to pass in element here, else its broken

Comment (erwinmombay) [inline extensions/amp-youtube/0.1/amp-youtube.js]: and pass through to super

Comment (cramforce) [inline extensions/amp-youtube/0.1/amp-youtube.js]: Good catch. Done.

Also: Yes, will need to declare all instance vars in constructor. Better for type checking and performance.

**Comment by erwinmombay** [comment]
@cramforce do let me know which extensions you are working on as i'll probably start on a few as well, especially the ones i own. (soundcloud, user-notification, amp-fresh, amp-live-list)

**Comment by erwinmombay** [comment]
LGTM

**Comment by cramforce** [comment]
@erwinmombay I wasn't planning to really do many. Maybe 1 or 2 more.

**Comment by aghassemi** [comment]
Cool, I will do the components I touch as well.

**Pull request merged**